gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/AbstractResource.java

```diff
@@ -109,34 +113,53 @@ protected boolean canReadAPIConfiguration() {
     }
 
     /**
-     * @return The list of API Membership for the authenticated user (direct membership or through groups)
+     * @return The list of API Membership for the authenticated user (direct membership or through groups or through primary owner groups)
      */
     private Stream<MembershipEntity> retrieveApiMembership() {
-        Stream<MembershipEntity> streamUserMembership = membershipService
+        Stream<MembershipEntity> streamUserAPIMembership = membershipService
                 .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), API).stream();
 
-        Stream<MembershipEntity> streamGroupMembership = membershipService
-                .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), GROUP).stream()
-                .filter(m -> m.getRoleId() != null && roleService.findById(m.getRoleId()).getScope().equals(RoleScope.API));
+        RoleEntity apiPrimaryOwnerRole = roleService.findByScopeAndName(RoleScope.API, SystemRole.PRIMARY_OWNER.name())
+                .orElseThrow(() -> new RoleNotFoundException(RoleScope.API, SystemRole.PRIMARY_OWNER.name()));
 
-        return Stream.concat(streamUserMembership, streamGroupMembership);
+        Stream<MembershipEntity> streamUserGroupMembership = membershipService
+                .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), GROUP).stream()
+                .filter(m -> {
+                    if (m.getRoleId() != null && !m.getRoleId().equals(apiPrimaryOwnerRole.getId())) {
+                        return roleService.findById(m.getRoleId()).getScope().equals(RoleScope.API);
+                    }
+                    return false;
+                });
+
+        Stream<MembershipEntity> streamUserPOGroupAPIMembership = membershipService
+                .getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), GROUP).stream()
```

Comment:
It seams to be the same call as the one in line 125-126.

Maybe we can extract the result of `membershipService.getMembershipsByMemberAndReference(USER, getAuthenticatedUser(), GROUP)` to avoid too many db calls ?